frontend/src/components/DynamicDiv/index.js

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import Button from "../Button";
+
+export default function Div({ style}) {
+  const [state, setState] = useState("Initial Text");
+  return (
+    <>
+      <div className="DynamicDiv" style={style}>
+        {state}
+      </div>
+      <Button
+        onClick={() => { 
+          setState("New text");
+        }}>
+        Change Text
+      </Button>
+    </>
+  );
+}
+
+Div.propTypes = {
+  style: PropTypes.object,
```

Comment:
I'm not sure why PropTypes for style needs to be object.  It doesn't seem like you utilized this in App.js.  Although I've never seen it before so I might not know exactly what it does.  I assume that it allows you to pass an object into the style prop, but you did something that can normally be done with your style prop in App.js.  Am I wrong?

Reply:
Think of this like defining a class.  I'm defining what kind of properties that Div can have, in this case, a style.  To set a style, I need to give it a style object, hence the PropTypes.object.  This is essentially setting the type of that property.

You're right that I didn't apply a style in App.js, but the important thing is that the option is there that way the Div can be used with custom styles, rather than being stuck with a single default one.  It is not needed for the Div to work function properly, it is an optional customization.